tests/function/test_config.py

```diff
@@ -29,50 +28,122 @@ def test_debug_in_virtwho_conf(self, virtwho, globalconf):
+        globalconf.update('global', 'debug', 'True')
+        globalconf.update('global', 'interval', '10')
+        result = virtwho.run_service()
+        assert (result['send'] == 1
+                and result['error'] == 0
+                and result['interval'] == 3600)
+
+        globalconf.update('global', 'interval', '60')
+        result = virtwho.run_service()
+        assert (result['send'] == 1
+                and result['error'] == 0
+                and result['interval'] == 60)
 
-    @pytest.mark.tier3
-    def test_oneshot_in_virtwho_conf(self):
-        """Just a demo
+        globalconf.update('global', 'interval', '120')
+        result = virtwho.run_service()
+        assert (result['send'] == 1
+                and result['error'] == 0
+                and result['interval'] == 120)
+
+    @pytest.mark.tier1
+    def test_oneshot_in_virtwho_conf(self, virtwho, globalconf):
+        """Test the oneshot option in /etc/virtwho.conf
 
         :title: virt-who: config: test oneshot option
-        :id: 15b45691-7358-4bc8-b49f-f87100bded1b
+        :id: 9e39f91f-80b5-4773-bef0-7facf8cb85e2
         :caseimportance: High
-        :tags: tier3
+        :tags: tier1
         :customerscenario: false
         :upstream: no
         :steps:
-            1.
+
+            1. Run virt-who with "oneshot=True" in /etc/virt-who.conf
+            2. Run virt-who with "oneshot=False" in /etc/virt-who.conf file
+
         :expectedresults:
-            1.
+
+            1. Can see 'Thread X stopped after running once' log in rhsm.log
+            2. Cannot see 'Thread X stopped after running once' log in rhsm.log
         """
-        logger.info("Succeeded to run the 'test_oneshot'")
+        globalconf.update('global', 'debug', 'True')
+        globalconf.update('global', 'oneshot', 'True')
+        result = virtwho.run_service()
+        assert (result['send'] == 1
+                and result['error'] == 0
+                and result['oneshot'] is True)
+
+        globalconf.update('global', 'oneshot', 'False')
+        result = virtwho.run_service()
+        assert (result['send'] == 1
+                and result['error'] == 0
+                and result['oneshot'] is False)
+
+    def test_print_in_virtwho_conf(self, virtwho, globalconf):
+        """Test the print_ option in /etc/virtwho.conf
+
+        :title: virt-who: config: test print_ option
+        :id: 25de8130-677f-43ca-b07d-a15f49e91205
+        :caseimportance: High
+        :tags: tier1
+        :customerscenario: false
+        :upstream: no
+        :steps:
+
+            1. Run virt-who with "print_=True" in /etc/virt-who.conf
+            2. Run virt-who with "print_=False" in /etc/virt-who.conf
+
+        :expectedresults:
+
+            1. the mappings send number and alive thread number of the virt-who is 0
+            2. the mappings send number and alive thread number of the virt-who is 1
+        """
+        globalconf.update('global', 'debug', 'True')
```

Comment:
It's better to test the 'print_=true' with `debug=True and False` to check if the [DEBUG] logs can be printed as configuration, and we should make sure the guest_id or some other keywords in the `result['print_json']` to make sure mapping is printed well.

Reply:
Added